executor/builder.go

```diff
@@ -1197,6 +1198,25 @@ func (b *executorBuilder) buildExplain(v *plannercore.Explain) Executor {
 		if b.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl == nil {
 			b.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl = execdetails.NewRuntimeStatsColl(nil)
 		}
+		// If the resource group name is not empty or "default", we could collect and display the RU
+		// runtime stats for analyze executor.
+		resourceGroupName := b.ctx.GetSessionVars().ResourceGroupName
+		if len(resourceGroupName) > 0 && resourceGroupName != variable.DefaultResourceGroupName {
+			// Try to register the RU runtime stats for analyze executor.
+			store, ok := b.ctx.GetStore().(interface {
+				CreateRURuntimeStats(uint64) *clientutil.RURuntimeStats
+			})
+			if ok {
+				// StartTS will be used to identify this SQL, so that the runtime stats could
```

Comment:
```suggestion
			if store, ok := b.ctx.GetStore().(interface {
				CreateRURuntimeStats(uint64) *clientutil.RURuntimeStats
			}); ok {
				// StartTS will be used to identify this SQL, so that the runtime stats could
```